fdbserver/ClusterController.actor.cpp

```diff
@@ -1446,7 +1453,29 @@ void checkOutstandingRecruitmentRequests( ClusterControllerData* self ) {
 	for( int i = 0; i < self->outstandingRecruitmentRequests.size(); i++ ) {
 		RecruitFromConfigurationRequest& req = self->outstandingRecruitmentRequests[i];
 		try {
-			req.reply.send( self->findWorkersForConfiguration( req ) );
+			RecruitFromConfigurationReply rep = self->findWorkersForConfiguration( req );
+			for(auto& it : rep.oldLogRouters) {
+				self->db.requiredAddresses.insert(it.address());
+				if( it.tLog.getEndpoint().addresses.secondaryAddress.present() ) self->db.requiredAddresses.insert(it.tLog.getEndpoint().addresses.secondaryAddress.get());
+			}
```

Comment:
Since this loop is written a bunch of times, it might be a good candidate for extracting into a function.